### PR TITLE
Feat: Migrate `query suggestions` module to composition api

### PR DESCRIPTION
**Pull request by andreadlgdo**
# Pull request template
 
Describe the purpose of the change, the specific changes done in detail, and the issue you have fixed.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [x] Open issue. If applicable, link: [EMP-4163](https://searchbroker.atlassian.net/browse/EMP-4163)

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [ ] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [ ] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

[EMP-4163]: https://searchbroker.atlassian.net/browse/EMP-4163?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/_vue3-migration-test/src/main.ts [hunk 1]
```diff
@@ -26,7 +27,14 @@ if (VUE_COMPAT_MODE === 2) {
   });
 }
 
-const adapter = {} as XComponentsAdapter;
+const adapter = {
+  querySuggestions: request => {
+    return new Promise(resolve => {
+      const suggestions = getQuerySuggestionsStub(request.query, 5);
+      resolve({ suggestions });
```
Comment (victorcg88) [inline packages/_vue3-migration-test/src/main.ts:34]:
```suggestion
      resolve({ getQuerySuggestionsStub(request.query, 5) });
```

File: packages/_vue3-migration-test/src/x-modules/query-suggestions/components/test-query-suggestions.vue [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  <SearchInput />
+  <QuerySuggestions />
+</template>
+
+<script setup>
+  import QuerySuggestions from '../../../../../x-components/src/x-modules/query-suggestions/components/query-suggestions.vue';
+  import SearchInput from '../../../../../x-components/src/x-modules/search-box/components/search-input.vue';
+</script>
+
+<style scoped></style>
```
Comment (victorcg88) [inline packages/_vue3-migration-test/src/x-modules/query-suggestions/components/test-query-suggestions.vue:11]: Style block could be deleted

```suggestion
```

File: packages/x-components/src/x-modules/query-suggestions/components/query-suggestion.vue [hunk 3]
```diff
@@ -32,37 +30,47 @@
...
+
+      /**
+       * Emits {@link QuerySuggestionsXEvents.UserSelectedAQuerySuggestion} with the suggestion as
+       * payload when selecting the query suggestion.
+       *
+       * @internal
+       */
+      const event = {
+        UserSelectedAQuerySuggestion: props.suggestion
+      };
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/query-suggestions/components/query-suggestion.vue:66]:
```suggestion
      const suggestionSelectedEvents = {
        UserSelectedAQuerySuggestion: props.suggestion
      };
```
so later on the template v-bind could be cleaner:

`v-bind="{ query, suggestion, suggestionSelectedEvents }"`

File: packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue [hunk 4]
```diff
@@ -49,20 +45,24 @@
...
+    xModule: querySuggestionsXModule.name,
+    setup: function () {
+      useRegisterXModule(querySuggestionsXModule);
+
+      /**
+       * The module's list of suggestions.
+       *
+       * @internal
+       */
+      const suggestions = useGetter('querySuggestions', ['querySuggestions']).querySuggestions;
```
Comment (victorcg88) [inline packages/x-components/src/x-modules/query-suggestions/components/query-suggestions.vue:61]:
```suggestion
      const {querySuggestions} = useGetter('querySuggestions', ['querySuggestions']);
```

**Review by victorcg88 — COMMENTED** [review]
I added some minor suggestions, but great job!

**Review by victorcg88 — APPROVED** [review]
LGTM!

**Review by victorcg88 — APPROVED** [review]

File: packages/_vue3-migration-test/package.json [hunk 5]
```diff
@@ -16,7 +16,7 @@
     "@vueuse/core": "~10.7.1",
     "vue": "^3.4.22",
     "vue-router": "^4.3.0",
-    "vuex": "^4.1.0"
+    "vuex": "4.0.2"
```
Comment (joseacabaneros) [inline packages/_vue3-migration-test/package.json:19]: Reactivity of vuex getters is broken in `4.1.0` when registering store modules dynamically. https://github.com/vuejs/vuex/issues/2217
It caused emitters didn't work when the trigger is a getter and not the state (`request` getter of QuerySuggestions x-module).
Downgraded to `4.0.2`

**Review by diegopf — APPROVED** [review]

**Pull request merged**